yt/visualization/base_plot_types.py

```diff
@@ -341,7 +341,15 @@ def _set_axes(self) -> None:
         self.image.axes.set_facecolor(self.colorbar_handler.background_color)
 
         self.cax.tick_params(which="both", direction="in")
-        self.cb = self.figure.colorbar(self.image, self.cax)
+        from mpl_toolkits.axes_grid1.axes_grid import CbarAxesBase
+
+        if isinstance(self.cax, CbarAxesBase):
+            # For creating a multipanel plot by ImageGrid
+            self.cb = self.figure.colorbar(
+                self.image, self.cax, location=self.cax.orientation
+            )
+        else:
+            self.cb = self.figure.colorbar(self.image, self.cax)
```

Comment:
Let's try minimizing the repetition if we can
```suggestion
            cb_kwargs = {"location": self.cax.orientation}
        else:
            cb_kwargs = {}
        self.cb = self.figure.colorbar(self.image, self.cax, **cb_kwargs)
```